WebDriverAgentLib/Routing/FBSession.m

```diff
@@ -178,8 +178,10 @@ - (XCUIApplication *)activeApplication
   if (nil != self.testedApplication) {
     XCUIApplicationState testedAppState = self.testedApplication.state;
     if (testedAppState >= XCUIApplicationStateRunningForeground) {
+      // We look for `SBTransientOverlayWindow` elements for half modals. See https://github.com/appium/WebDriverAgent/pull/946
+      NSPredicate *searchPredicate = [NSPredicate predicateWithFormat:@"%K == %@ OR %K == %@", @"elementType", @(XCUIElementTypeAlert), @"identifier", @"SBTransientOverlayWindow"];
       if ([FBConfiguration shouldRespectSystemAlerts]
-          && [XCUIApplication.fb_systemApplication descendantsMatchingType:XCUIElementTypeAlert].count > 0) {
+          && [[XCUIApplication.fb_systemApplication descendantsMatchingType:XCUIElementTypeAny] matchingPredicate:searchPredicate].count > 0) {
```

Comment:
same here

Reply:
Thank you! Added new lines as https://github.com/appium/WebDriverAgent/pull/946/commits/bb33df8a900d5599e0dc25f61bead6a12e211547